packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -49,6 +49,7 @@ abstract class ProgressIndicator extends StatefulWidget {
     this.valueColor,
     this.semanticsLabel,
     this.semanticsValue,
+    this.hasRoundStroke,
```

Comment:
Since StrokeCap.round is only relevant to CircularProgressIndicator (If the Linear one chooses a radius instead), should this only be on the CircularProgressIndicator class?

Reply:
I am also not sure if the name is clear as a user, why not use StrokeCap directly?

Reply:
StrokeCap has 3 values and only 2 of them are relevant. Butt and Square collide with each other, because Circular uses Butt when it has value, Square when it doesn't (it is spinning).

Reply:
> should this only be on the CircularProgressIndicator

Probably. How do you think the theme should be? `CircularHasRoundIndicator` and `LinearIndicatorRadius`?

Reply:
Renamed to `preferRoundIndicator`.

Reply:
> only 2 of them are relevant

I disagree with this statement, why are there 3 then? :) It's not for us to decide that one is irrelevant, the uses cases have been pointed out in the discussion of this PR.

> How do you think the theme should be? CircularHasRoundIndicator and LinearIndicatorRadius?

Are there not individual component themes for circular and linear? I would not put both in the super class ProgressIndicatorTheme.

Reply:
> I disagree with this statement, why are there 3 then? :) It's not for us to decide that one is irrelevant, the uses cases have been pointed out in the discussion of this PR.

If it is square, the `value` is represented wrong: https://github.com/flutter/flutter/issues/109137#issuecomment-1209391300.
If it is butt, it is represented correctly.

But if it is butt, on indeterminate, it "disappears", so whoever implemented it made it square so it is always visible.

Exposing this to the user is hard: on square they will have the wrong value, on butt they will have an "unintended" animation (which, if it is to be customizable, we could allow changing other things, such as speed).

> Are there not individual component themes for circular and linear?

No, I could make one, I guess..

Reply:
> No, I could make one, I guess..

Oh, I meant that if they don't exist then we should probably omit theming support for now. :) No worries, no need to create more themes right now for this change.